src/TensorOneD.cpp

```diff
@@ -0,0 +1,139 @@
+                                   GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+        if (input.array_items().size() == 0) {
+            throw geopm::Exception("Empty array is invalid for neural network weights.\n",
+                                   GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+
+        set_dim(input.array_items().size());
+
+        for (std::size_t idx = 0; idx < m_vec.size(); idx++) {
+            if (!input[idx].is_number()) {
+                throw geopm::Exception("Non-numeric type found in neural network weights.\n",
+                                       GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+            }
+            m_vec[idx] = input[idx].number_value();
+        }
+    }
+
+    std::size_t TensorOneD::get_dim() const
+    {
+        return m_vec.size();
+    }
+
+    void TensorOneD::set_dim(std::size_t dim)
+    {
+        m_vec.resize(dim);
+    }
+
+    TensorOneD TensorOneD::operator+(const TensorOneD& other)
+    {
+        if (get_dim() != other.get_dim()) {
+            throw geopm::Exception("Adding vectors of mismatched dimensions.",
+                                   GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+
+        TensorOneD rval(m_vec.size());
+        std::transform(m_vec.begin(), m_vec.end(), other.m_vec.begin(), rval.m_vec.begin(), std::plus<float>());
+
+        return std::move(rval);
+    }
+
+    TensorOneD TensorOneD::operator-(const TensorOneD& other)
+    {
+        if (get_dim() != other.get_dim()) {
+            throw geopm::Exception("Subtracting vectors of mismatched dimensions.",
+                                   GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+
+        TensorOneD rval(m_vec.size());
+        std::transform(m_vec.begin(), m_vec.end(), other.m_vec.begin(), rval.m_vec.begin(), std::minus<float>());
+
+        return std::move(rval);
+    }
+
+
+    float TensorOneD::operator*(const TensorOneD& other)
+    {
+        if (get_dim() != other.get_dim()) {
+            throw geopm::Exception("Inner product of vectors of mismatched dimensions.",
+                                   GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+
+        return std::move(std::inner_product(m_vec.begin(), m_vec.end(), other.m_vec.begin(), 0));
+    }
+
+    TensorOneD& TensorOneD::operator=(const TensorOneD &other)
+    {
+        m_vec = other.m_vec;
+        return *this;
+    }
+
+    TensorOneD& TensorOneD::operator=(TensorOneD &&other)
+    {
+        m_vec = std::move(other.m_vec);
+        return *this;
+    }
+
+    float& TensorOneD::operator[] (std::size_t idx)
+    {
+        return m_vec.at(idx);
+    }
+
+    float TensorOneD::operator[] (std::size_t idx) const
+    {
+        return m_vec.at(idx);
+    }
+
+    TensorOneD TensorOneD::sigmoid() const
+    {
+        TensorOneD rval(m_vec.size());
+        for(std::size_t idx = 0; idx < m_vec.size(); idx++) {
+            rval[idx] = 1/(1 + expf(-(m_vec.at(idx))));
```

Comment:
Seems like there's potential for a divide-by-zero error here if the denominator evaluates to 0.  Is this possible?  Should error checks be added here?